### PR TITLE
Authentication + Homepage 

### PR DESCRIPTION
**Pull request by tejas-muthya37**
The Screens folder in src contains the code for Authentication and Homepage of the app.

### CODE REVIEW

File: src/Context/address-context.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      return {
+        ...state,
+        address: { ...state.address, addressLine2: action.payload },
+      };
+    case "address-line-3":
+      return {
+        ...state,
+        address: { ...state.address, addressLine3: action.payload },
+      };
+    case "address-line-4":
```
Comment (SJTGSHIVAM) [inline src/Context/address-context.js:22]: I am not sure if this is the best way to do it but try to use better names than line one two .... THis defiantly depends on the context so if it works and makes semantic sense then it sfine. (just a suggestion let me know what you think about it)

File: src/Context/address-context.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      return {
+        ...state,
+        address: { ...state.address, addressLine7: action.payload },
+      };
+    case "edit-address":
+      return {
+        ...state,
+        address: {
+          addressLine1: action.payload[0],
+          addressLine2: action.payload[1],
```
Comment (SJTGSHIVAM) [inline src/Context/address-context.js:47]: Its always better to share paylaod with same shape as required at the other end

Comment (SJTGSHIVAM) [reply src/Context/address-context.js:47]: if done so this code can be reduced to something like {...state,{...address{...action.payload}}}

Comment (SJTGSHIVAM) [reply src/Context/address-context.js:47]: I believe this is something worth the effort. please take this up in a later sprint as per the time permits

File: src/Context/filter-context.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          : (item.show = true);
+        return true;
+      });
+      return {
+        ...state,
+        items: state.items,
+      };
+    case "Category filter":
+      if (state.categoryFiltersFlag === false) {
+        state.items.map((item) => {
```
Comment (SJTGSHIVAM) [inline src/Context/filter-context.js:52]: I am not sure why we are using a map if we are not storing output here.

File: src/Context/filter-context.js [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+          item.categoryName === action.payload.id
+            ? (item.show = true)
+            : (item.show = false);
```
Comment (SJTGSHIVAM) [inline src/Context/filter-context.js:55]: yep ?: is good but we should always prefer if else construct when intention in to perform an action not to return an expression. This makes code more readable. 
(just a suggestion)

File: src/Context/filter-context.js (hunk 3 above)
Comment (SJTGSHIVAM) [reply src/Context/filter-context.js:52]: forEach is a good alternative

**Review by Ronak-Mutha — APPROVED** [review]

File: src/Context/address-context.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+const AddressContext = createContext();
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "address-line-1":
+      return {
+        ...state,
+        address: { ...state.address, addressLine1: action.payload },
+      };
+    case "address-line-2":
```
Comment (ramalingaa) [inline src/Context/address-context.js:12]: Try using specific naming convention ex: locality, city

File: src/Context/filter-context.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
+import { createContext, useContext, useReducer } from "react";
+
+const FilterContext = createContext();
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case "Setup":
```
Comment (ramalingaa) [inline src/Context/filter-context.js:7]: I suggest not to use generice words for naming, setup doesn't convey anything to reader

**Review by ramalingaa — APPROVED** [review]
Code pattern is clean, one suggestion would be please focus on naming variables
Everything else looks good

**Review by SJTGSHIVAM — APPROVED** [review]
Overall code looks good. There are some minor suggestions added which can be taken into account in a later sprint. Right now this PR looks good. Great job Tejas!

File: src/Context/filter-context.js (hunk 6 above)
Comment (SJTGSHIVAM) [reply src/Context/filter-context.js:7]: Hey Ramalinga 
Can you interpret more on this?

Comment (tejas-muthya37) [reply src/Context/filter-context.js:7]: @ramalingaa - Yes, I should have written something like 'Videos Array setup'. Will do the necessary changes, thanks.

**Pull request merged**